python/lsst/obs/decam/ingestCalibs.py

```diff
@@ -135,7 +143,12 @@ def getExtensionName(md):
         result : `str`
             The string from the EXTNAME header card.
         """
-        return md.getScalar('EXTNAME')
+        try:
+            # This returns a tuple
+            ext = md.getScalar("EXTNAME")
+            return ext
+        except KeyError:
+            return None
```

Comment:
Use `md.get("EXTNAME", None)` as if it was a normal dict. Then you don't need the try/except.

Reply:
When I don't override it, the code in pipe_tasks' ingest throws a KeyError because `md.getScalar("EXTNAME")` doesn't returns something that is indexable with `[1]` (I pasted the pipe_tasks code below). I can't quite figure out what is happening here.
```
    @staticmethod
    def getExtensionName(md):
        """ Get the name of an extension.
        @param md: PropertySet like one obtained from lsst.afw.fits.readMetadata)
        @return Name of the extension if it exists.  None otherwise.
        """
        try:
            # This returns a tuple
            ext = md.getScalar("EXTNAME")
            return ext[1]
        except lsst.pex.exceptions.Exception:
            return None
```

Reply:
What happens if you use my suggestion of removing all of that and using `md.get("EXTNAME", None)`?

The related question is why do these raws you are ingesting not have an EXTNAME?

Reply:
if `ext` is returning a scalar this implies that the EXTNAME string only has one character in it???  When I first looked I assumed it was reading the second element from a list but getScalar doesn't return a list. What is EXTNAME in this file?

Reply:
The EXTNAMEs are things like 'S29       ', 'N4       ', etc. for stack-built MasterCals. It's the CP MasterCals which don't have EXTNAME at all, because they're multi-extension FITS files.

Reply:
`ext = md.get("EXTNAME", None)` seems to work fine, but I'm loathe to put that in pipe_tasks and break non-decam things.

Reply:
EXTNAME is part of the FITS standard so there's no problem with looking for it. The problem seems to be in requiring the value of EXTNAME to be something useful. I still don't understand how S29 is going to do anything other than return "2" -- is that helpful? That `ext[1]` makes no sense to me at all.

Reply:
Sorry I was mixing up errors here. There are two problems. (1) the pipe_tasks getExtension behavior fails for stack-built calibs because all but the 0th extension lack EXTNAME (it checks every extension). (2) agreed the pipe_tasks getExtension `ext[1]` makes no sense. As best as I can tell the getExtension method was only ever added because of decam. I'll fix this in pipe_tasks.